tools/gradle-plugin/build.gradle.kts

```diff
@@ -0,0 +1,79 @@
+plugins {
+    alias(libs.plugins.kotlin.jvm)
+    alias(libs.plugins.buildConfig)
+    `java-gradle-plugin`
+}
+
+tasks.validatePlugins {
+    enableStricterValidation = true
+}
+
+val originUrl = providers
+    .exec { commandLine("git", "remote", "get-url", "origin") }
+    .standardOutput
+    .asText
+    .map { it.trim() }
+
+gradlePlugin {
+    vcsUrl = originUrl
+    website = originUrl
+
+    plugins {
+        create("valkyrie") {
+            id = "io.github.composegears.valkyrie"
+            displayName = name
+            implementationClass = "io.github.composegears.valkyrie.gradle.ValkyrieGradlePlugin"
+            description = "Generates Kotlin accessors for ImageVectors, based on input SVG files"
+            tags.addAll("kotlin", "svg", "xml", "imagevector", "valkyrie")
+        }
+    }
+}
+
+buildConfig {
+    generateAtSync = true
+    sourceSets.getByName("test") {
+        packageName = "io.github.composegears.valkyrie.gradle"
+        useKotlinOutput {
+            internalVisibility = true
+            topLevelConstants = true
+        }
+
+        // TODO: Set up test matrix for different gradle/agp/kotlin version combos?
+        buildConfigField("AGP_VERSION", libs.gradle.agp.get().version)
+        buildConfigField("KOTLIN_VERSION", libs.gradle.kotlin.get().version)
+        buildConfigField("GRADLE_VERSION", "8.14.1")
+
+        fun Project.sharedTestResourcesDir(name: String) = project
+            .project(projects.components.test.path)
+            .layout
+            .projectDirectory
+            .dir("sharedTestResources/imagevector/$name")
+            .asFile
+            .absolutePath
+
+        // So we can copy the shared test SVG/XML files into our test cases
+        buildConfigField("SAMPLE_SVG_DIR", sharedTestResourcesDir("svg"))
+        buildConfigField("SAMPLE_XML_DIR", sharedTestResourcesDir("xml"))
+    }
+}
+
+dependencies {
+    compileOnly(gradleApi())
+    compileOnly(gradleKotlinDsl())
+
+    implementation(libs.gradle.agp)
+    implementation(libs.gradle.kotlin)
```

Comment:
:warning: Warning :warning: 

these might affect downstream projects using earlier versions of either of these - but I couldn't get the unit tests to run properly without them. Setting to `compileOnly` normally works for gradle plugins like this, but I think I may be missing something